sbol_utilities/component.py

```diff
@@ -1,44 +1,78 @@
 from __future__ import annotations
 
-from itertools import chain
-from typing import Dict, Iterable, List, Union, Set, Optional, Tuple
+from typing import Dict, Iterable, List, Union, Optional, Tuple
 
 import sbol3
 import tyto
-from rdflib import URIRef
 
-from sbol_utilities.helper_functions import id_sort, find_child
+from sbol_utilities.helper_functions import id_sort, find_child, find_top_level, SBOL3PassiveVisitor, cached_references
 from sbol_utilities.workarounds import get_parent
 
 
 # TODO: consider allowing return of LocalSubComponent and ExternallyDefined
-def contained_components(roots: Union[sbol3.TopLevel, Iterable[sbol3.TopLevel]]) -> Set[sbol3.Component]:
+def contained_components(roots: Union[sbol3.TopLevel, Iterable[sbol3.TopLevel]]) \
+        -> set[sbol3.Component]:
```

Comment:
If it were me I would break the line before the closing paren to avoid the backslash:

```suggestion
def contained_components(roots: Union[sbol3.TopLevel, Iterable[sbol3.TopLevel]]
        ) -> set[sbol3.Component]:
```

That might also need some reindenting, not sure.

Reply:
I realized this line didn't actually need a wrap with the current style limit